src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -630,9 +643,11 @@ private void processLogsAndUpload() throws InterruptedException {
                 } catch (SecurityException e) {
                     logger.atError().cause(e).log("Unable to get log files for {} from {}",
                             componentName, componentLogConfiguration.getDirectoryPath());
+                } catch (InvalidLogGroupException e) {
+                    logger.atError().cause(e).log("Unable to read the directory");
                 }
             }
-            if (componentLogFileInformation.get().isPresent()) {
+            if (!isActiveFileCompleted.get() && componentLogFileInformation.get().isPresent()) {
```

Comment:
Do we need this new check?  Would the active file be in the componentLogFileInformation already?